src/data/providers/balancer-api/modules/pool-state/index.ts

```diff
@@ -186,7 +186,7 @@ export class Pools {
         const { data } = await this.balancerApiClient.fetch({
             query: this.poolStateQuery,
             variables: {
-                id,
+                id: id.toLowerCase(),
```

Comment:
I didn't know that was an issue. I'd say it's ok to have it as a quick fix here, but ideally the API should support non-lowercase inputs as well. I'll create an issue on their repo.